web/src/components/Upload/util.ts

```diff
@@ -1,43 +1,115 @@
+export const getGpsStart = (textLines: Array<string>): moment.Moment => {
+  /* Return moment under all circumstances.
+  This enables validity checks on returned moment */
+  for (const line of textLines) {
+    if (line.startsWith('$GPRMC')) {
+      const lineFields: Array<string> = line.split(',', 10)
+      const startTime: string = lineFields[1]
+      const startDate: string = lineFields[9]
+      const startDateTime: moment.Moment = moment.utc(
+        `${startDate} ${startTime}`,
+        'DDMMYYYY hh mm ss.SS'
+      )
+      return startDateTime
+    }
+  }
+  return moment.utc('')
+}
+
+export const getDustrakStart = (textLines: Array<string>): moment.Moment => {
+  if (textLines.length > 7) {
+    const startTime: string = textLines[6].split(',')[1]
+    const startDate: string = textLines[7].split(',')[1]
+    const startDatetime: moment.Moment = moment.tz(
+      `${startDate} ${startTime}`,
+      'MM-DD-YYYY hh:mm:ss a',
+      'America/Los_Angeles'
+    )
+    return startDatetime
+  } else {
+    return moment('')
+  }
+}
 
-const parseDateFromLogFileName = (fileName: string): string => {
-  const fileNameArr = fileName.split('_')
-  const dateString = fileNameArr[1]
-  return `${dateString.slice(0, 4)}/${dateString.slice(
-    4,
-    6
-  )}/${dateString.slice(6)}`
+export const getDustrakEnd = (
+  textLines: Array<string>,
+  startDatetime: moment.Moment
+): moment.Moment => {
+  if (textLines.length > 8) {
+    const endDatetime: moment.Moment = startDatetime.clone()
+    const testLength: string = textLines[8].split(',')[1]
+    const [days, hours, minutes]: Array<string> = testLength.split(':')
+    endDatetime
+      .add(days, 'days')
+      .add(hours, 'hours')
+      .add(minutes, 'minutes')
+    return endDatetime
+  } else {
+    return moment('')
+  }
 }
 
-export const validateFiles = (files: Array<FileWithPath>): string => {
+export const identFiles = (
+  files: Array<FileWithPath>
+): Array<File | undefined> => {
+  let logFile: File | undefined
+  let csvFile: File | undefined
+
+  files.forEach(file => {
+    if (file.name.endsWith('.csv')) {
+      csvFile = file
+    }
+
+    if (file.name.endsWith('.log')) {
+      logFile = file
+    }
+  })
+  return [logFile, csvFile]
+}
+
+export const validateFiles = async (
+  files: Array<FileWithPath>
+): Promise<string> => {
   if (files.length > 0 && files.length < 2) {
     return 'We need one GPS log file and one DusTrak cvs file. Please add a file to continue.'
   }
-
   if (files.length > 2) {
     return 'We need exactly one GPS log file and one DusTrak cvs file. Please remove additional files.'
   }
-
   if (files.length === 2) {
-    let logFile: FileWithPath | undefined
-    let csvFile: FileWithPath | undefined
-
-    files.forEach(file => {
-      if (file.name.endsWith('.csv')) {
-        csvFile = file
-      }
-
-      if (file.name.endsWith('.log')) {
-        logFile = file
-      }
-    })
-
+    const [logFile, csvFile] = identFiles(files)
```

Comment:
High-level comment - sometimes the files are distinguished using terms 'csv' and 'log' (like here) and sometimes 'gps' and 'dustrak' (elsewhere). It would be better to use the same naming scheme and make it fairly use case agnostic (e.g. 'gps'/'sensor' or 'location'/'sensor').

Reply:
...yeah, I couldn't decide which one made more sense. I chose the worst path and used both.

Reply:
sorry to clarify, @nbilenko, would you prefer that these files be called `gps` and `dustrak` to keep it consistent with the backend?

Reply:
I'd be fine with either, as long as it's consistent - going with `gps` and `dustrak` would indeed make it so we don't have to change it on the backend, so that seems easier!

Reply:
Issue #221